ompi/mca/coll/base/coll_base_reduce_scatter_block.c

```diff
@@ -91,17 +93,38 @@ ompi_coll_base_reduce_scatter_block_basic_linear(const void *sbuf, void *rbuf, i
         recv_buf = recv_buf_free - gap;
     }
 
-    /* reduction */
-    err =
-        comm->c_coll->coll_reduce(sbuf, recv_buf, count, dtype, op, 0,
-                                 comm, comm->c_coll->coll_reduce_module);
+    if ( OPAL_UNLIKELY(count > INT_MAX) ) {
```

Comment:
This approach assumes the root has all the memory necessary to allocate the entire array. I don't think this is a reasonable approach. Instead, I would do a loop around reduce with the size of a peer, and then a send to that particular peer. This will reduce the memory requirements on the root.

Reply:
That sounds reasonable. I'll work on that this week and update the PR.

Reply:
I just pushed a new version that addresses this improvement. Take a look and let me know what you think.